rir/src/compiler/analysis/scope.cpp

```diff
@@ -72,8 +76,13 @@ void TheScopeAnalysis::apply(AS& envs, Instruction* i) const {
     if (mk) {
         Value* parentEnv = mk->env();
         // If we know the caller, we can fill in the parent env
-        if (parentEnv == Env::theContext() && invocation)
-            parentEnv = invocation->env();
```

Comment:
this is actually a bugfix. In

`mkEnv(parent=parent?)` we currently plug in the env of the caller as the parent, instead of the env of the declaration site :(